Mlem/Views/Tabs/Profile/AvatarBannerView.swift

```diff
@@ -0,0 +1,108 @@
+        self.banner = community.banner
+        @AppStorage("shouldShowCommunityHeaders") var shouldShowCommunityHeaders: Bool = true
+        self.showBanner = shouldShowCommunityHeaders
+        @AppStorage("shouldShowCommunityIcons") var shouldShowCommunityIcons: Bool = true
+        self.showAvatar = shouldShowCommunityIcons
+    }
+    
+    static let bannerHeight: CGFloat = 170
+    static let avatarOverdraw: CGFloat = 40
+    static let avatarSize: CGFloat = 108
+    static let avatarPadding: CGFloat = AppConstants.postAndCommentSpacing
+    
+    var avatarView: some View {
+        AvatarView(
+            url: avatar,
+            type: type,
+            avatarSize: AvatarBannerView.avatarSize,
+            lineWidth: 0,
+            iconResolution: .unrestricted
+        )
+    }
+    
+    var body: some View {
+        Group {
+            if let banner, showBanner {
+                ZStack(alignment: .bottom) {
+                    VStack {
+                        LazyImage(url: banner) { state in
+                            VStack {
+                                if let image = state.image {
+                                    image
+                                        .resizable()
+                                        .aspectRatio(contentMode: .fill)
+                                        .clipped()
+                                } else {
+                                    Color.secondarySystemBackground
+                                }
+                            }
+                            .frame(minWidth: 0, maxWidth: .infinity)
+                            .frame(height: AvatarBannerView.bannerHeight)
+                            .clipped()
+                            .clipShape(RoundedRectangle(cornerRadius: AppConstants.largeItemCornerRadius))
+                            .mask {
+                                ZStack(alignment: .bottom) {
+                                    Color.black
+                                    if showAvatar {
+                                        Circle()
+                                            .frame(
+                                                width: AvatarBannerView.avatarSize + AvatarBannerView.avatarPadding * 2,
+                                                height: AvatarBannerView.avatarSize + AvatarBannerView.avatarPadding * 2
+                                            )
+                                            .offset(y: AvatarBannerView.avatarOverdraw + AvatarBannerView.avatarPadding)
+                                            .blendMode(.destinationOut)
+                                    }
+                                }
+                                .compositingGroup()
+                            }
+                            
+                        }
+                        Spacer()
+                    }
+                    if showAvatar {
+                        avatarView
+                    }
```

Comment:
*nit*

Probably nicer to rip out the `ZStack` and put this in an `.overlay`--I think SwiftUI handles that a little more efficiently because it saves a size computation

```suggestion
                    .overlay {
                        if showAvatar {
                            avatarView
                                .frame(maxHeight: .infinity, alignment: .bottom)
                        }
                    }
```